Add Create a channel tutorial without a system channel and using test network

pamandrejko opened the pull request:
Signed-off-by: pama-ibm <pama@ibm.com>

Because we have now updated the Fabric sample Test Network to no longer use a system channel, we need the following doc updates:
- Added a new Create a channel without a system channel tutorial using the Test Network.
- The new tutorial is largely based on the legacy create a channel tutorial (with a system channel) except that it uses the new`TwoOrgsApplicationGenesis` profile in the test network configtx.yaml and uses the `osnadmin channel join` command to create the channel. 
- Because it no longer works with the latest version of the test network, I removed the legacy Create a channel tutorial that is based on the test network, and link to it in the Fabric v2.2 documentation instead.
- Added a link from the existing Create a channel without a system channel tutorial to the new tutorial for users who prefer to try it out with the test network.
- Updated the Create a Channel tutorial Overview topic to include the new tutorial. 
- As a side effort, I also removed 2 topics from the Fabric FAQs which mentioned the system channel.

#### Type of change
- Documentation update

#### Description

<!--- Describe your changes in detail, including motivation. -->

#### Additional details

<!--- Additional implementation details or comments to reviewers. -->
<!--- Summarize how the pull request was tested (if not obvious from commit). -->

#### Related issues

I also need to open another PR against the Fabric v2.2 Create a channel tutorial which instructs them to download to Fabric v2.2 Samples once the release is cut.

pamandrejko committed:
Add Create a channel tutorial without a system channel and using test network

Signed-off-by: pama-ibm <pama@ibm.com>

tock-ibm commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+### Consenter vs. Follower
+
+Notice the ordering node was joined to the channel with a `consensusRelation: "consenter"`. If you ran the command against an ordering node that is not included in the list of `Consenters:` in the `configtx.yaml` file (or the channel configuration consenter set), it is added to the channel as a `follower`. This is important because anytime a new orderer joins the cluster, you need to ensure that  ...
```
just a typo: `is available / can be used`

pamandrejko replied:
Thanks. Changed it to `can be used`

nikhil550 commented:
I like the new tutorial,  but I find the new information flow a bit confusing. The create a channel using the test network doesn't just cover the test network. It covers how to create a channel in general, and uses the test network as an example, like the other tutorials in the docs. It also covers many of the same information covered in the how to create a channel without a system channel tutorial. Instead, the new doc is more "tutorial" like, using the test network as an example rather than covering all scenarios and details.

My suggestion would be to rename the tutorial simply, `creating a channel`, replacing the old tutorial with the new method. The creating a channel without a system channel could then be made an operations guide, perhaps focusing on the removing the system channel or migration.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+Notice that the peers are running on ports `7051` and `9051`, while the orderer is running on port `7050`. We will use these ports in subsequent commands.  
+
+By default, when you start the test network, it does not contain any channels. The following instructions demonstrate how to add a channel that is named `channel1` to this network.
```
I would mention, either here or up above when `network.sh up` is run, that there is an option to automatically `createChannel` when bringing up the test network, and that this tutorial doesn't do that so that you can learn how to create one manually.  Otherwise the reader may think that they always have to run through these steps to work with a channel on test network.

pamandrejko replied:
Added this to the first paragraph:
It is worth noting that the test network includes a `createChannel` subcommand that can be used to create a channel, but this tutorial explains how do it manually, the process that is required when you do not use the test network.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+### Set up the configtxgen tool
+
+Channels are created by building a channel creation transaction and submitting the transaction to the ordering service. The channel creation transaction specifies the initial configuration of the channel and is used by the ordering service to write the channel genesis block. While it is possible to build the channel creation transaction file manually, it is easier to use the [configtxgen](../comm ...
```
```suggestion
Channels are created by generating a channel creation transaction in a genesis block, and then passing that genesis block to an ordering service node in a join request. The channel creation transaction specifies the initial configuration of the channel and can be created by the [configtxgen](../commands/configtxgen.html) tool. The tool reads the test network `configtx\configtx.yaml` file that defines the configuration of our channel, and then writes the relevant information into the channel creation transaction and outputs a genesis block including the channel creation transaction. When you [installed Fabric](install.html), the `configtxgen` tool was installed in the `fabric-samples\bin` directory for you.
```

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+### The configtx.yaml file
+
+The `configtx.yaml` file specifies the **channel configuration** of new channels. The information that is required to build the channel configuration is specified in a readable and editable form in the `configtx.yaml` file. The `configtxgen` tool uses the channel profiles that are defined in the `configtx.yaml` file to create the channel configuration and write it to the [protobuf format](https:/ ...
```
This paragraph duplicates some of the information in the first paragraph of the prior section.

pamandrejko replied:
I've tried to remove the redundancies.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+You can find the `configtx.yaml` file that is used to deploy the test network in the `configtx` folder in the `test-network/configtx` directory. The file contains the following information that we will use to create our new channel:
+
+- **Organizations:** The organizations that can become members of your channel. Each organization has a reference to the cryptographic material that is used to build the [channel MSP](../membership/membership.html).
```
Mention that this includes ordering organizations and peer organizations.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+You can find the `configtx.yaml` file that is used to deploy the test network in the `configtx` folder in the `test-network/configtx` directory. The file contains the following information that we will use to create our new channel:
+
+- **Organizations:** The organizations that can become members of your channel. Each organization has a reference to the cryptographic material that is used to build the [channel MSP](../membership/membership.html).
+- **Ordering service:** Which ordering nodes will form the ordering service of the network, and consensus method they will use to agree to a common order of transactions. This section also defines the ordering nodes that are part of the ordering service consenter set. In the test network sample, there is only a single ordering node, but in a production network we recommend **five** ordering nodes ...
```
The first two sentences essentially say the same thing and can be combined.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+- **Ordering service:** Which ordering nodes will form the ordering service of the network, and consensus method they will use to agree to a common order of transactions. This section also defines the ordering nodes that are part of the ordering service consenter set. In the test network sample, there is only a single ordering node, but in a production network we recommend **five** ordering nodes ...
+    ```
+    Addresses:
+        - orderer.example.com:7050
````
The Addresses entry should be removed from here, and in fact from the test network configtx.yaml file itself.
We now recommend to only specify the OrdererEndpoints in the org section above, you can see it here:
https://github.com/hyperledger/fabric-samples/blob/master/test-network/configtx/configtx.yaml#L44-L45

pamandrejko replied:
I did not know that. Removed it from the tutorial.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+          ServerTLSCert: ../organizations/ordererOrganizations/example.com/orderers/orderer.example.com/tls/server.crt
+    ```
+- **Channel policies** Different sections of the file work together to define the policies that will govern how organizations interact with the channel and which organizations need to approve channel updates. For the purposes of this tutorial, we will use the default policies used by Fabric.
+- **Channel profiles** Each channel profile references information from other sections of the `configtx.yaml` file to build a channel configuration. The profiles are used to create the genesis block of application channel. Notice that the `configtx.yaml` file in the test network includes a single profile named `TwoOrgsApplicationGenesis` that we will use to generate the channel.
````
```suggestion
- **Channel profiles** Each channel profile references information from other sections of the `configtx.yaml` file to build a channel configuration. The profiles are used to create the genesis block of application channel. Notice that the `configtx.yaml` file in the test network includes a single profile named `TwoOrgsApplicationGenesis` that we will use to generate the create channel transaction.
```

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+                <<: *ApplicationCapabilities
+    ```
+
+The profile includes both peer organizations, `Org1` and `Org2` as well as the ordering organization `OrdererOrg`. Additional ordering nodes and ordering organizations can be added or removed from the consenter set using channel updates.
````
```suggestion
The profile includes both peer organizations, `Org1` and `Org2` as well as the ordering organization `OrdererOrg`. Additional ordering nodes and ordering organizations can be added or removed from the consenter set at a later time using a channel update transaction.
```

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+Want to learn more about this file and how to build your own channel application profiles? Visit [Using configtx.yaml to create a channel genesis block](create_channel_config.html) tutorial for more details. For now, we will return to the operational aspects of creating the channel, though we will reference parts of this file in future steps.
+
+## Step one: Create the channel genesis block
```
Keep consistent with the other create channel tutorial, it called Step 1:
Generate the genesis block of the channel

pamandrejko replied:
I changed it to Step one: Generate the genesis block of the channel

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+Because we have started the Fabric test network, we are ready to create a new channel for our peer organizations. We have already set the environment variables that are required to use the `configtxgen` tool. To simplify subsequent commands, we also need to set some environment variables to establish the locations of the certificates for the nodes in the test network:
+
+```
+export CORE_PEER_TLS_ENABLED=true
````
Set this one later in the peer join section.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+export CORE_PEER_TLS_ENABLED=true
+export ORDERER_CA=${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem
+export PEER0_ORG1_CA=${PWD}/organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt
+export PEER0_ORG2_CA=${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt
```
Remove these two, the path can be provided later in the peer join section when CORE_PEER_TLS_ROOTCERT_FILE is set for each peer.
Since each is only set once in the peer section, the additional env variables across the two sections just adds to the complexity with an unnecessary indirection.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+export PEER0_ORG2_CA=${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt
+export ORDERER_ADMIN_TLS_SIGN_CERT=${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/tls/server.crt
+export ORDERER_ADMIN_TLS_PRIVATE_KEY=${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/tls/server.key
+```
````
For these two and ORDERER_CA, set them later right above the osnadmin command.  It is easier to mentally keep track of the env variables if they are set right before use.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+## Step one: Create the channel genesis block
+
+Because we have started the Fabric test network, we are ready to create a new channel for our peer organizations. We have already set the environment variables that are required to use the `configtxgen` tool. To simplify subsequent commands, we also need to set some environment variables to establish the locations of the certificates for the nodes in the test network:
```
```suggestion
Because we have started the Fabric test network, we are ready to create a new channel. We have already set the environment variables that are required to use the `configtxgen` tool. To simplify subsequent commands, we also need to set some environment variables to establish the locations of the certificates for the nodes in the test network:
```

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+### Consenter vs. Follower
+
+Notice the ordering node was joined to the channel with a `consensusRelation: "consenter"`. If you ran the command against an ordering node that is not included in the list of `Consenters:` in the `configtx.yaml` file (or the channel configuration consenter set), it is added to the channel as a `follower`. This is important because anytime a new orderer joins the cluster, you need to ensure that  ...
```
Isn't this information elsewhere in the docs? Couldn't we link to it instead of describe the whole thing again here? Would nice to keep the test network tutorial short and sweet for newbies. I think it is worth mentioning in a sentence or two, and then link to the details.

pamandrejko replied:
This comment reinforces the need for another conceptual topic or ops guide content. I left 2 sentences in and then linked to where we currently explain more on this.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
```diff
@@ -0,0 +1,444 @@
+
+### Join peers to the channel
+
+The test network includes two peer organizations each with one peer. But before we can use the peer CLI, we need to set some environment variables to specify which peer we are acting as. Set the following environment variables to indicate that we are acting as the peer in `Org1`.
```
```suggestion
The test network includes two peer organizations each with one peer. But before we can use the peer CLI, we need to set some environment variables to specify which user (client MSP) we are acting as and which peer we are targeting. Set the following environment variables to indicate that we are acting as the Org1 admin and targeting the Org1 peer.
```

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+FABRIC_CFG_PATH=$PWD/../config/
+```
+
+To join the test network peer from `Org1` to the channel `channel1` simply run:
````
```suggestion
To join the test network peer from `Org1` to the channel `channel1` simply pass the genesis block in a join request:
```

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+Version: 1.0, Sequence: 1, Endorsement Plugin: escc, Validation Plugin: vscc, Approvals: [Org1MSP: true, Org2MSP: true]
+Query chaincode definition successful on peer0.org2 on channel 'channel1'
+```
+Then run the following command to initialize the ledger:
````
```suggestion
Then run the following command to initialize some assets on the ledger:
```

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+[chaincodeCmd] chaincodeInvokeOrQuery -> INFO 001 Chaincode invoke successful. result: status:200
+```
+
+Confirm the data was added by issuing the following query:
````
```suggestion
Confirm the assets were added to the ledger by issuing the following query:
```

denyeart reviewed (commented):
I agree with Nik that it is a bit awkward to have two tutorials for creating a channel. Since most of the other tutorials in the Tutorials section target the test network, I would keep this new test network tutorial in the Tutorials section, and split the content from the other one into a Concept topic, a 'migrating from system channel' topic, and deployment guide topic. If I recall correctly, that was always the plan, the content was simply all jammed together to speed delivery for the v2.3 docs. That being said, it can be done later. For now as a first step, I'm fine with the two tutorial approach that you have.

pamandrejko commented:
I think we all are in agreement that we need an additional Conceptual and/or Operations Guide topic that discusses more of the details around this. Until that exists, we need both tutorials, because the original tutorial we created for the system channel removal work is the only place we currently describe in detail how to configure an ordering node, considerations around joining additional ordering nodes to the channel, and removing the system channel from existing ordering nodes. I started a jira for that consolidation work here: https://jira.hyperledger.org/browse/FAB-18411

In the meantime, for readability, I went ahead and removed the text `without a system channel` from the titles of both tutorials and updated all links throughout our documentation that included the tutorial names.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+
+```
+export CORE_PEER_TLS_ENABLED=true
+export PEER0_ORG1_CA=${PWD}/organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt
````
No need to set env variable PEER0_ORG1_CA.
Just set the path in CORE_PEER_TLS_ROOTCERT_FILE directly.
Unnecessary indirection only adds confusion.

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+
+We repeat these steps for the `Org2` peer. Set the following environment variables to operate the `peer` CLI as the `Org2` admin. The environment variables will also set the `Org2` peer, ``peer0.org2.example.com``, as the target peer.
+```
+export PEER0_ORG2_CA=${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt
````
Remove

denyeart commented on docs/source/create_channel/create_channel_test_net.md:
````diff
@@ -0,0 +1,444 @@
+```
+export PEER0_ORG2_CA=${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt
+export CORE_PEER_LOCALMSPID="Org2MSP"
+export CORE_PEER_TLS_ROOTCERT_FILE=${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt
````
This one is good!

pamandrejko replied:
OK. Modified all of these to:
```
export CORE_PEER_TLS_ENABLED=true
export CORE_PEER_LOCALMSPID="Org1MSP"
export CORE_PEER_TLS_ROOTCERT_FILE=${PWD}/organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt
export CORE_PEER_MSPCONFIGPATH=${PWD}/organizations/peerOrganizations/org1.example.com/users/Admin@org1.example.com/msp
export CORE_PEER_ADDRESS=localhost:7051
```

and 
```
export CORE_PEER_LOCALMSPID="Org2MSP"
export CORE_PEER_TLS_ROOTCERT_FILE=${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt
export CORE_PEER_MSPCONFIGPATH=${PWD}/organizations/peerOrganizations/org2.example.com/users/Admin@org2.example.com/msp
export CORE_PEER_ADDRESS=localhost:9051
```

denyeart reviewed (dismissed):
LGTM now. Sounds like @lesleyannjordan is going to try it out.

pamandrejko commented:
I addressed the following comments from @lesleyannjordan 

- "Channel profile Each channel"  is this meant to be on new line and bulleted?  only showing as bold
- This sentence "The process for setting an anchor peer is similar to the steps that are required to make other channel updates and provides an introduction to how to use configtxlator to update a channel configuration." under anchor peer doesn't seem to read well to me.
- Would it be good to mention the jq install is required up in the pre-req section?
- Is "The new protobuf named channel_update.pb contains the anchor peer" is it channel_update.pb or config_update.pb?
- In paragraph "Now that the channel has been updated by adding two channel configuration blocks to the channel genesis block, the height of the channel will have grown to three:"  is it worth mentioning that you expect the Hashes to be different to the example shown below?

denyeart commented:
@Mergifyio backport release-2.3
